### PR TITLE
Update use of various deprecated APIs

### LINKED ISSUE #4720: Communicator: Use `close` instead of deprecated `stop`
This is from the [warnings of the test suite](https://github.com/aiidateam/aiida-core/pull/4719/checks?check_run_id=1858668311#step:9:558)

```
  /home/runner/work/aiida-core/aiida-core/aiida/cmdline/commands/cmd_status.py:116: UnsupportedWarning: stop is unsupported as of 0.7.0. Use close() instead.  Unlike stop(), close() is permanent and the class should not be used again after this.
    comm.stop()
```

@sphuber Could you check whether these calls can simply be replaced or whether you are using them in a way that one expects it to be possible to reopen the communicator again?

### LINKED ISSUE #4721: Several tests write to `node.open`
E.g. 
```python
    def test_node_repo_cat(self):
        """Test 'verdi node repo cat' command."""
        # Test cat binary files
        with self.folder_node.open('filename.txt.gz', 'wb') as fh_out:
            fh_out.write(gzip.compress(b'COMPRESS'))
```

Resulting in warnings
```
tests/cmdline/commands/test_calcjob.py::TestVerdiCalculation::test_calcjob_inputcat
tests/cmdline/commands/test_calcjob.py::TestVerdiCalculation::test_calcjob_outputcat
tests/cmdline/commands/test_node.py::TestVerdiNode::test_node_repo_cat
tests/engine/test_calc_job.py::test_parse_not_implemented
tests/engine/test_calc_job.py::test_parse_scheduler_excepted
  /home/runner/work/aiida-core/aiida-core/aiida/orm/nodes/node.py:452: AiidaDeprecationWarning: from v2.0 only the modes 'r' and 'rb' will be accepted
    warnings.warn("from v2.0 only the modes 'r' and 'rb' will be accepted", AiidaDeprecationWarning)  # pylint: disable=no-member
```

How should these tests be rewritten?

### PR DESCRIPTION
**Pull request by ltalirz**
The current test suite raises [946 warnings](https://github.com/aiidateam/aiida-core/runs/1856480956?check_suite_focus=true#step:9:790).

Having a go at reducing those a bit so that we can actually learn something from warnings again.

fixes #4721 (node.open used in write mode) via commit c6ebd3a8a6779c74d86ebd30c10febf890d63046
fixes #4720 (communicator: stop() => close() cdf759b)

### CODE REVIEW

**Comment by giovannipizzi** [comment]
I just rebased as the jenkins tests were failing, and #4565 changes them.

The error was
```
--> Action: 'converge'
    
    PLAY [Set up AiiDa Environment] ************************************************
    
    TASK [reentry scan] ************************************************************
    Tuesday 09 February 2021  10:26:20 +0000 (0:00:00.040)       0:00:00.040 ******
fatal: [molecule-aiida-sqlalchemy-89d03cac-ba18-4e91-b452-af8d018bbe24]: FAILED! => changed=false
  cmd:
  - /opt/conda/bin/reentry
  - scan
  delta: '0:00:00.393638'
  end: '2021-02-09 10:26:21.254565'
  msg: non-zero return code
  rc: 1
  start: '2021-02-09 10:26:20.860927'
  stderr: |-
    Traceback (most recent call last):
      File "/opt/conda/bin/reentry", line 5, in <module>
        from reentry.cli import reentry
      File "/opt/conda/lib/python3.7/site-packages/reentry/__init__.py", line 2, in <module>
        from reentry.default_manager import DEFAULT_MANAGER as manager
      File "/opt/conda/lib/python3.7/site-packages/reentry/default_manager.py", line 32, in <module>
        class PluginManager(object):
      File "/opt/conda/lib/python3.7/site-packages/reentry/default_manager.py", line 35, in PluginManager
        def __init__(self, backend=get_default_backend(), scan_for_not_found=True):
      File "/opt/conda/lib/python3.7/site-packages/reentry/default_manager.py", line 29, in get_default_backend
        return JsonBackend(datafile=get_datafile())
      File "/opt/conda/lib/python3.7/site-packages/reentry/jsonbackend.py", line 24, in __init__
        self._epmap = self.read()
      File "/opt/conda/lib/python3.7/site-packages/reentry/jsonbackend.py", line 35, in read
        return json.load(cache_file_obj)
      File "/opt/conda/lib/python3.7/json/__init__.py", line 296, in load
        parse_constant=parse_constant, object_pairs_hook=object_pairs_hook, **kw)
      File "/opt/conda/lib/python3.7/json/__init__.py", line 348, in loads
        return _default_decoder.decode(s)
      File "/opt/conda/lib/python3.7/json/decoder.py", line 337, in decode
        obj, end = self.raw_decode(s, idx=_w(s, 0).end())
      File "/opt/conda/lib/python3.7/json/decoder.py", line 355, in raw_decode
        raise JSONDecodeError("Expecting value", s, err.value) from None
    json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
  stderr_lines: <omitted>
  stdout: ''
  stdout_lines: <omitted>
    
    PLAY RECAP *********************************************************************
    molecule-aiida-sqlalchemy-89d03cac-ba18-4e91-b452-af8d018bbe24 : ok=0    changed=0    unreachable=0    failed=1    skipped=0    rescued=0    ignored=0
    
    Playbook run took 0 days, 0 hours, 0 minutes, 1 seconds
    Tuesday 09 February 2021  10:26:21 +0000 (0:00:01.073)       0:00:01.113 ******
    ===============================================================================
    reentry scan ------------------------------------------------------------ 1.07s
```
This happens (I believe) when both tests (SQLA + Django) work at the same time and change the JSON file, as they share the same "environment" on Jenkins, differently from GH actions where each test is running in a different "container".

@chrisjsewell could you please confirm that this error has been fixed by your changes in #4565?
Or, if not, this might be a randomly-occurring error, so probably it would be good to open an issue to fix it (it will randomly occur again in the future).

File: tests/tools/importexport/__init__.py
Comment (giovannipizzi) [inline tests/tools/importexport/__init__.py:24]: Shouldn't you put it back to the correct level in the tear down class method? Otherwise other tests, later, using the same logger might not output relevant warnings

**Review by giovannipizzi — DISMISSED** [review]

**Comment by giovannipizzi** [comment]
Sorry, converting back to draft as I plan to fix one more set of things (deprecations warnings of tests of deprecated features)

**Comment by chrisjsewell** [comment]
> Or, if not, this might be a randomly-occurring error, so probably it would be good to open an issue to fix it (it will randomly occur again in the future).

added to https://github.com/aiidateam/aiida-core/issues/4718#issuecomment-775870496

**Comment by giovannipizzi** [comment]
Weird, I pushed one more commit but I don't see it? [This one](https://github.com/ltalirz/aiida-core/commit/53ed4bb3e810b8e6728ca9a7b1ea942512a674ce). Maybe a GitHub temporary issue or did I do something wrong?

**Comment by giovannipizzi** [comment]
@ltalirz Let's see if this commit pops up... anyway, for me this PR is ready to go (if tests pass). Great if you could give a review to my last commits before I merge (I think I still need to give the final review, but I don't want to self-review just because you opened the PR).

**Comment by ltalirz** [comment]
Thanks @giovannipizzi , I reviewed your commits cdc1cf6 and 7403dbc, which look fine

**Comment by giovannipizzi** [comment]
Thanks! Did you also review 53ed4bb3e810? and do you know why it's not popping up here?

**Comment by giovannipizzi** [comment]
I'll try rebasing in the hope things start working again. Note that commit hashes will change

**Comment by giovannipizzi** [comment]
Weird... anyway now it works. @ltalirz great if you could review also e5bf51b0528c7364950ed08e14a7d0513e0841be; I will then approve and merge (assuming that I didn't break some test...)

**Review by giovannipizzi — APPROVED** [review]

**Comment by giovannipizzi** [comment]
Thanks @ltalirz for the final review. I pre-approved this PR, so that if tests pass you can just merge it (I'll need to to away from my computer for a bit).

**Comment by ltalirz** [comment]
Just as a final tally: now down to 714 warnings  (from 946) on the sqla python 3.8 build
(and that is after re-enabling `ResourceWarnings`, not all of which were fixed)

**Comment by giovannipizzi** [comment]
Thanks!
Indeed, and the majority of those should be addressed in #4722

**Pull request merged**
